internal/access/group.go

```diff
@@ -30,13 +33,29 @@ func isUserInGroup(c *gin.Context, requestedResourceID uid.ID) (bool, error) {
 	return false, nil
 }
 
-func ListGroups(c *gin.Context, name string) ([]models.Group, error) {
+func ListGroups(c *gin.Context, name string, userID uid.ID) ([]models.Group, error) {
+	var selectors []data.SelectorFunc
+	if name != "" {
+		selectors = append(selectors, data.ByName(name))
+	}
```

Comment:
this is what data.ByOptionalName is for. move this logic into the selector.

Reply:
Not sure I agree with that. It cleans up the callers quite a bit, and the caller needs to explicitly think about whether or not the parameter could be optional or not, as it introduces security problems if a required argument is treated as optional.